test/integration/roomMembersTest.ts

```diff
@@ -580,18 +616,29 @@ describe("Test: Testing RoomMemberManager", function() {
                 assert.ok(usersInWitnessRoom.includes(userId), `After a ${experiment.name}, good user ${userId} should still be in witness room`);
                 assert.ok(usersInUnprotectedWitnessRoom.includes(userId), `After a ${experiment.name}, good user ${userId} should still be in unprotected witness room`);
             }
-            for (let userId of badUserIds) {
-                assert.ok(!usersInRoom.includes(userId), `After a ${experiment.name}, bad user ${userId} should NOT be in affected room`);
-                assert.equal(usersInWitnessRoom.includes(userId), !experiment.shouldAffectWitnessRoom, `After a ${experiment.name}, bad user ${userId} should ${experiment.shouldAffectWitnessRoom ? "NOT" : "still"} be in witness room`);
-                assert.ok(usersInUnprotectedWitnessRoom.includes(userId), `After a ${experiment.name}, bad user ${userId} should still be in unprotected witness room`);
-                const leaveEvent = await this.mjolnir.client.getRoomStateEvent(roomId, "m.room.member", userId);
-                switch (experiment.method) {
-                    case Method.kick:
-                        assert.equal(leaveEvent.membership, "leave");
-                        break;
-                    case Method.ban:
-                        assert.equal(leaveEvent.membership, "ban");
-                        break;
+            if (experiment.method === Method.mute) {
```

Comment:
do we need to check unmuting these people will work afterwards?

Reply:
We probably should, yeah.